ipfs-cluster-service/migrate.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"bufio"
+
+	"fmt"
+	"os"
+
+	"github.com/urfave/cli"
+
+	peer "github.com/libp2p/go-libp2p-peer"
+	
+	ipfscluster "github.com/ipfs/ipfs-cluster"
+	"github.com/ipfs/ipfs-cluster/state/mapstate"
+	"github.com/ipfs/ipfs-cluster/consensus/raft"
+)
+
+
+func upgrade(c *cli.Context) error {
+	if c.NArg() < 1 || c.NArg() > 2 {
+		return fmt.Errorf("Usage: <BACKUP-FILE-PATH> [RAFT-DATA-DIR]")
+	}
+	//Load configs                                                             
+	cfg, clusterCfg, _, _, consensusCfg, _, _, _ := makeConfigs()
+	err := cfg.LoadJSONFromFile(configPath)
+	if err != nil {
+		return err
+	}
+	backupFilePath := c.Args().First()
+	var raftDataPath string
```

Comment:
I'm not super sold on letting the user provide a custom raftDataPath with all the logic it adds and needing to carry it down 3 levels of functions. perhaps just set consensusCfg.DataFolder to the value of raftDataPath?

 Note that consensusCfg.DataFolder is normally empty and that in that case, baseDir+defaultDataSubfolder should be used. That's mostly for the raft component's functions.

Reply:
Thanks, given the above info I'm happy using the default or config value.